templates/vue/src/components/NodeModal.vue

```diff
@@ -206,35 +218,47 @@ export default {
         this.videoLoaded
       )
     },
+    nodeIdNumber() {
+      return Number(this.nodeId)
+    },
+    tabIndex: {
+      get() {
+        const tabIndex = tabOrdering.findIndex(t => t === this.currentTab)
+        return this.currentTab === "" || tabIndex === -1 ? 0 : tabIndex
+      },
+      set(newIndex) {
+        this.currentTab = tabOrdering[newIndex]
+        this.$router.push(
+          "/nodes/" + this.modalType + "/" + this.nodeId + "/" + this.currentTab
+        )
+      },
+    },
+  },
+  watch: {
+    $route(to, from) {
+      if (this.currentTab !== to.params.tab && to !== from) {
+        this.currentTab = to.params.tab
+      }
+    },
+  },
```

Comment:
I think rather than doing this we should read the active tab directly from the current route. We can do this through the `active` prop in `b-tab` and by listening to the `@click` event also emitted by `b-tab`. This makes it so that the single source of truth is the current route, and we don't have to manage state inside `NodeModal`.

This also means that we have to link directly to the tab that we want to go to, e.g. pressing the add button would link to `/add/1234/content` instead of just `/add/1234`.

The docs for the tab component could be a good starting point for this: https://bootstrap-vue.org/docs/components/tabs

I hope that makes sense! If not, DM me and we can talk about it more.

Reply:
I think that you make a really good point here, so I removed the parts where we keep track of the `currentTab` and deal with storing that state, and instead refer directly to the route. Take a look at the changes to the NodeModal file and let me know if that's what you had in mind